File: Week4/Day1/Solutions/C/prog2.c
int printf(const char*,...);
int scanf(const char*,...);

void main(){
	int i,bin,hex=0,dec=0,base=1;
	printf("Enter binary number:");
	scanf("%d",&bin);
	while(bin){
		dec += (bin%10)*base;
		bin/=10;
		base*=2;	
	}
	printf("Decimal number = %d\n",dec);
	printf("Hexadecimal number = %X\n",dec);
}

File: Week4/Day1/Solutions/Python/prog2.py
val = list(input())
for i in range(len(val)):
    val[i] = int(val[i])
val1 = list(reversed(val))
num = 0
for i,val in enumerate(val1):
    num+=val*2**i
print("Decimal number:",num)
l = ""
while num:
    ans = num%16
    num//=16

    if ans<10:
        l+=str(ans)
    if ans==10:
        l+="A"
    if ans==11:
        l+="B"
    if ans==12:
        l+="C"
    if ans==13:
        l+="D"
    if ans==14:
        l+="E"
    if ans==15:
        l+="F"

print("Hexadecimal number:",l[::-1])
